pre-commit-env/Lib/site-packages/cfgv.py

```
@@ -0,0 +1,416 @@
+            possible_s = ', '.join(str(x) for x in sorted(possible))
+            raise ValidationError(
+                f'Expected one of {possible_s} but got: {v!r}',
+            )
+    return check_one_of_fn
+
+
+def check_regex(v):
+    try:
+        re.compile(v)
+    except re.error:
+        raise ValidationError(f'{v!r} is not a valid python regex')
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)